R/fn-heatwave.R

```diff
@@ -0,0 +1,393 @@
+
+    d <- dplyr::bind_rows(d1, d2, d3)
+    d <- dplyr::group_by(d, stn_id, year, measure)
+    arrow::write_dataset(d, parquet_path, format = "parquet")
+  })
+
+  attr(parquet_path, "timestamp") <- as.character(Sys.time())
+  parquet_path
+}
+
+get_ahccd_stations <- function() {
+  stations_url <- "https://crd-data-donnees-rdc.ec.gc.ca/CDAS/products/EC_data/AHCCD_daily/Homog_Temperature_Stations_Gen3.xls"
+
+  stations_file <- normalizePath(file.path("data", "ahccd_sources", basename(stations_url)),
+                                 mustWork = FALSE)
+  dir.create(dirname(stations_file), showWarnings = FALSE)
+
+  download.file(stations_url, destfile = stations_file, method = "curl")
+
+  colnames <- names(readxl::read_excel(stations_file, skip = 2, n_max = 2))
+
+  stations <- readxl::read_excel(stations_file, skip = 4, col_names = colnames)
+
+  stations <- stations |>
+    janitor::clean_names() |>
+    dplyr::mutate(from = paste(from, x6, sep = "-"),
+                  to = paste(to, x8, sep = "-")) |>
+    dplyr::select(-x6, -x8)
+
+  sf::st_as_sf(stations, coords = c("long_deg", "lat_deg"), crs = 4326)
+}
+
+get_bc_target_stations <- function(stations, buffer, crs) {
+  bc_buff <- sf::st_buffer(st_union(bcmaps::bc_bound()), dist = buffer * 1000)
+  stations <- sf::st_transform(stations, sf::st_crs(bc_buff))
+  stations <- sf::st_intersection(stations, bc_buff, sparse = FALSE)
+  sf::st_transform(stations, crs)
+}
+
+get_dem <- function(aoi = NULL, res) {
+  if (is.null(aoi)) {
+    aoi <- bcmaps::bc_bound() |>
+      sf::st_buffer(50000) |>
+      sf::st_transform(4326)
+  }
+
+  vrt <- bcmaps::cded(aoi, check_tiles = FALSE, ask = FALSE)
+  dem_stars <- stars::read_stars(vrt)
+  dem_stars_out <- stars::st_warp(
+    dem_stars,
+    stars::st_as_stars(sf::st_bbox(aoi), dx = res),
+    method = "bilinear", use_gdal = TRUE
+  )
+  fname <- paste0("data/dem_stars_", as.character(res), ".tif")
+  stars::write_stars(dem_stars_out, fname)
+  stars::read_stars(fname, proxy = FALSE)
+}
+
+model_temps_xyz <- function(temp_data, stations, months) {
+  temp_data <- temp_data |>
+    dplyr::filter(month %in% months)
+
+  stations <- cbind(sf::st_drop_geometry(stations),
+                    sf::st_coordinates(stations)) |>
+    dplyr::select(stn_id, x = X, y = Y, elevation = elev_m)
+
+  days <- unique(temp_data$date)
+
+  out <- future.apply::future_lapply(days, function(d) {
+    data <- temp_data[temp_data$date == d, , drop = FALSE]
+    df <- dplyr::left_join(stations, data, by = "stn_id") |>
+      dplyr::select(x, y, elevation, temp)
+    temp_tps(df)
+  }, future.seed = 13L)
+
+  stats::setNames(out, as.character(days))
+}
+
+temp_tps <- function(df) {
+  df <- na.omit(df)
+  indep <- df[, c("x", "y", "elevation")]
+
+  fields::Tps(x = indep, Y = df$temp, miles = FALSE)
+}
+
+interpolate_daily_temps <- function(model_list, dem, variable) {
+  stars_list <- future.apply::future_lapply(model_list, function(mod) {
+    predict(dem, mod)
```

Comment:
consider explicit namespaces

Reply:
We can't namespace `predict()` because it is an `S3` method (`predict.stars()`)